### PR TITLE
Upgrade to Realm Core v11.0.3

### LINKED ISSUE #3785: Upgrade to Realm Core v11.0.3

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

This closes #3785 

It also includes the key based change set for dictionaries

## ☑️ ToDos

* [x] 📝 Changelog entry
* [x] 📝 `Compatibility` label is updated or copied from previous entry
* [x] 🚦 Tests
* ~[ ] 📝 Public documentation PR created or is not necessary~
* ~[ ] 💥 `Breaking` label has been applied or is not necessary~

*If this PR adds or changes public API's:*
* [ ] typescript definitions file is updated
* [ ] jsdoc files updated
* ~[ ] Chrome debug API is updated if API is available on React Native~

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,26 +1,25 @@
 x.x.x Release notes (yyyy-MM-dd)
 =============================================================
-NOTE: Realm file format is likely to break and you CAN NOT revert back to the previous file format - DATA WILL BE LOST!!!!
-NOTE: Sync protocol version 4: CANNOT SYNC WITH MONGODB REALM CLOUD.
-NOTE: Bump file format version to 21. NO DOWNGRADE PATH IS AVAILABLE.
+NOTE: Bump file format version to 22. NO DOWNGRADE PATH IS AVAILABLE.
 
 ### Enhancements
-* None.
+* Change listener for dictionary is not key based.
```
Comment (fronck) [inline CHANGELOG.md:6]: Shouldn't this be noted under a "known issues" heading or similar?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,26 +1,25 @@
...
 * Observing a dictionary holding links to objects would crash. ([realm/realm-core#4711](https://github.com/realm/realm-core/issues/4711), since v11.0.0-beta.1)
+* Better support for Android 12. ([realm/realm-core#4666](https://github.com/realm/realm-core/issues/4666))
 
 ### Compatibility
 * MongoDB Realm Cloud.
-* APIs are backwards compatible with all previous releases of Realm JavaScript in the 10.x.y series.
-* File format: generates Realms with format 21 (reads and upgrades file format v5 or later for non-synced Realm, upgrades file format v10 for synced Realms).
+* Realm Studio v11.0.0.
+* APIs are backwards compatible with all previous releases of Realm JavaScript in the 10.5.x series.
+* File format: generates Realms with format 22 (reads and upgrades file format v5 or later for non-synced Realm, upgrades file format v10 for synced Realms).
```
Comment (fronck) [inline CHANGELOG.md:17]:
```suggestion
* File format: generates Realms with format 22 (reads and upgrades file format v5 or later for non-synced Realm, upgrades file format v10 or later for synced Realms).
```

File: scripts/changelog-header.sh [hunk 3]
```diff
@@ -15,11 +15,14 @@ x.x.x Release notes (yyyy-MM-dd)
 
 ### Compatibility
 * MongoDB Realm Cloud.
-* APIs are backwards compatible with all previous releases of Realm JavaScript in the 10.x.y series.
-* File format: generates Realms with format v20 (reads and upgrades file format v5 or later for non-synced Realm, upgrades file format v10 for synced Realms).
+* Realm Studio v11.0.0
+* APIs are backwards compatible with all previous releases of Realm JavaScript in the 10.5.y series.
+* File format: generates Realms with format v22 (reads and upgrades file format v5 or later for non-synced Realm, upgrades file format v10 for synced Realms).
```
Comment (fronck) [inline scripts/changelog-header.sh:20]:
```suggestion
* File format: generates Realms with format v22 (reads and upgrades file format v5 or later for non-synced Realm, upgrades file format v10 or later for synced Realms).
```

File: scripts/changelog-header.sh [hunk 4]
```diff
@@ -15,11 +15,14 @@ x.x.x Release notes (yyyy-MM-dd)
...
 
 ### Internal
-* None.
+* <Either mention core version or upgrade>
+* <Using Realm Core vX.Y.Z>
+* <Upgraded Realm Core from vX.Y.Z to vA.B.C>
```
Comment (fronck) [inline scripts/changelog-header.sh:25]:
```suggestion
* None.
```

File: CHANGELOG.md (hunk 1 above)
Comment (kraenhansen) [reply CHANGELOG.md:6]: My guess is that he meant to write "now" instead of "not":
```suggestion
* Change listener for dictionary is now key based.
```

**Review by kraenhansen — APPROVED** [review]
Looks about right to me 🙂

File: CHANGELOG.md (hunk 1 above)
Comment (bmunkholm) [reply CHANGELOG.md:6]: Add link to PR or issue (that has a link to PR)

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,26 +1,25 @@
...
 
 ### Enhancements
-* None.
+* Change listener for dictionary is not key based.
 
 ### Fixed
-
 * Performance regression for some scenarios of writing/creating objects with a primary key. ([realm/realm-core#4522](https://github.com/realm/realm-core/issues/4522))
 * Observing a dictionary holding links to objects would crash. ([realm/realm-core#4711](https://github.com/realm/realm-core/issues/4711), since v11.0.0-beta.1)
+* Better support for Android 12. ([realm/realm-core#4666](https://github.com/realm/realm-core/issues/4666))
```
Comment (bmunkholm) [inline CHANGELOG.md:11]: What does "better" mean?

File: scripts/changelog-header.sh (hunk 4 above)
Comment (bmunkholm) [reply scripts/changelog-header.sh:25]: @fronck Why would you want to remove this from the template?

File: CHANGELOG.md (hunk 1 above)
Comment (bmunkholm) [reply CHANGELOG.md:6]: Perhaps also mention that this is breaking the API introduced in beta "?"?

File: scripts/changelog-header.sh (hunk 4 above)
Comment (fronck) [reply scripts/changelog-header.sh:25]: Woops, wrong file.  Good catch 👍

File: CHANGELOG.md (hunk 1 above)
Comment (kneth) [reply CHANGELOG.md:6]: I think I need to rephrase it to "Change listeners for dictionaries are now key based"

File: CHANGELOG.md (hunk 5 above)
Comment (kneth) [reply CHANGELOG.md:11]: Let me change it to "Async callbacks not triggered on Android 12 emulator"

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,26 +1,25 @@
...
 
 ### Compatibility
 * MongoDB Realm Cloud.
-* APIs are backwards compatible with all previous releases of Realm JavaScript in the 10.x.y series.
-* File format: generates Realms with format 21 (reads and upgrades file format v5 or later for non-synced Realm, upgrades file format v10 for synced Realms).
+* Realm Studio v11.0.0.
+* APIs are backwards compatible with all previous releases of Realm JavaScript in the 10.5.x series.
+* File format: generates Realms with format 22 (reads and upgrades file format v5 or later for non-synced Realm, upgrades file format v10 for synced Realms).
 
 ### Internal
```
Comment (fronck) [inline CHANGELOG.md:19]:
```suggestion
### Known Issues
 * Set iterators do not work under Jest (#3788).

### Internal
```

**Review by fronck — APPROVED** [review]
LGTM with a small addendum.

File: CHANGELOG.md (hunk 6 above)
Comment (bmunkholm) [reply CHANGELOG.md:22]: It would be new if we start to track known issues in the release notes in general. And it would be hard to know where to stop? We have a bunch of known issues as github issues :-). Perhaps we could mention this limitation with the set documentation?
